res/controllers/engine-api.d.ts

```diff
@@ -0,0 +1,279 @@
+     */
+    function makeUnbufferedConnection(group: string, name: string, callback: CoCallback): ScriptConnection | undefined;
+
+    /**
+     * This function is a legacy version of makeConnection with several alternate
+     * ways of invoking it. The callback function can be passed either as a string of
+     * JavaScript code that evaluates to a function or an actual JavaScript function.
+     *
+     * @param group Group of the control e.g. "[Channel1]"
+     * @param name Name of the control e.g. "VuMeter"
+     * @param callback JS function, which will be called every time, the value of the connected control changes.
+     * @param disconnect If "true", all connections to the ControlObject are removed. [default = false]
+     * @returns Returns script connection object on success, otherwise 'undefined' or 'false' depending on the error cause.
+     * @deprecated Use {@link makeConnection} instead
+     */
+    function connectControl(group: string, name: string, callback: CoCallback, disconnect?: boolean): ScriptConnection | boolean | undefined;
+
+
+    /**
+     * Triggers the execution of all connected callback functions, with the actual value of a control.
+     * Note: To trigger a single connection, us {@link ScriptConnection.trigger} instead
+     *
+     * @param group Group of the control e.g. "[Channel1]"
+     * @param name Name of the control e.g. "play_indicator"
+     */
+    function trigger(group: string, name: string): void;
+
+    /** @deprecated Use {@link console.log} instead */
+    function log(message: string): void;
+
+    type TimerID = number;
+
+    /**
+     * Starts a timer that will call the specified script function
+     *
+     * @param interval Time in milliseconds until the function is executed.
+     *                 Intervals below 20ms are not allowed.
+     * @param scriptCode Function to be executed,
+     *                   you can also use closures as:
+     *                   function() { print("Executed Timer") }
+     * @param oneShot If true the function is only once,
+     *                if false the function is executed repeatedly  [default = false]
+     * @returns timerId which is needed to stop a timer.
+     *          In case of an error, 0 is returned.
+     */
+    function beginTimer(interval: number, scriptCode: () => any, oneShot?: boolean): TimerID;
+
+    /**
+     * Stops the specified timer
+     *
+     * @param timerId ID of the timer
+     */
+    function stopTimer(timerId: TimerID): void;
+
+    /**
+     * Jogwheel function to be called when scratching starts (usually when the wheel is touched)
+     * This function contains an parametrizeable alpha-beta filter, which influences the
+     * responsiveness and looseness of the imaginary slip mat
+     *
+     * @param deck The deck number to use, e.g: 1
+     * @param intervalsPerRev The resolution of the MIDI control (in intervals per revolution, typically 128.)
```

Comment:
I think this varies widely between different controllers, I'd take it out otherwise developers might just blindly use the default instead of actually measuring the value for their hardware.
```suggestion
     * @param intervalsPerRev The resolution of the MIDI control (in intervals per revolution)
```